### PR TITLE
implement current user status

### PR DESCRIPTION
**Pull request by Mana12011207**
I implemented the current-user endpoint to fetch the login user status. 

Screenshot 2023-12-12 at 23 53 33

### CODE REVIEW

File: src/controllers/AuthController.js [hunk 1]
```diff
@@ -183,3 +183,11 @@ export const resetPassword = catchAsync(async (req, res, next) => {
   // Send success response
   res.status(200).json({ message: 'Password has been successfully reset' });
 });
+
+export const currentUser = async (req, res) => {
```
Comment (irene2mana) [inline src/controllers/AuthController.js:187]: I think you might need to `import catchAsync from '../utils/catchAsync.js'` and use it before async() to catch any errors that occur. However, let's wait for today's meeting so we can hear Ellen's explanation first, which will help us gain a better understanding.

File: src/controllers/AuthController.js [hunk 2]
```diff
@@ -183,3 +183,13 @@ export const resetPassword = catchAsync(async (req, res, next) => {
   // Send success response
   res.status(200).json({ message: 'Password has been successfully reset' });
 });
+
+// Attempt to find a user in the database using the user ID.
+export const currentUser = catchAsync(async (req, res) => {
```
Comment (irene2mana) [inline src/controllers/AuthController.js:188]: `catchAsync` seems not to work without the `next` parameter, so you must add it after `res` to make the function work.

**Review by irene2mana — CHANGES_REQUESTED** [review]

File: src/controllers/AuthController.js (hunk 2 above)
Comment (irenenguyen1017) [inline src/controllers/AuthController.js:188]: `catchAsync` seems not to work without the `next` parameter, so you must add it after `res` to make the function work.

**Review by irenenguyen1017 — CHANGES_REQUESTED** [review]

File: src/controllers/AuthController.js (hunk 2 above)
Comment (ellenpham) [reply src/controllers/AuthController.js:188]: yes, you will have to add "next" to the params.

**Review by ellenpham — APPROVED** [review]

**Review by irenenguyen1017 — APPROVED** [review]

**Pull request merged**